src/app/pages/services/services.component.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+import { ServiceName, serviceNames } from '../../enums/service-name.enum';
+
+const fakeDataSource: Service[] = [...serviceNames.keys()]
+  .filter((service) => ![ServiceName.Gluster, ServiceName.Afp].includes(service))
+  .map((service, id) => {
+    return {
+      id,
+      service,
+      state: ServiceStatus.Stopped,
+      enable: false,
+    } as Service;
+  });
+
+describe('ServicesComponent', () => {
+  let spectator: SpectatorRouting<ServicesComponent>;
+  let loader: HarnessLoader;
+  let ws: WebSocketService;
+
+  const createComponent = createRoutingFactory({
+    component: ServicesComponent,
+    imports: [
+      CoreComponents,
+      EntityModule,
+      IxTableModule,
+      FormsModule,
+    ],
+    providers: [
+      mockWebsocket([
+        mockCall('service.query', fakeDataSource),
+        mockCall('service.update'),
+        mockCall('service.start'),
+        mockCall('service.stop'),
+      ]),
+      mockProvider(DialogService),
+      mockProvider(IxSlideInService),
+      mockProvider(CoreService),
+      mockProvider(IscsiService),
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createComponent();
+    loader = TestbedHarnessEnvironment.loader(spectator.fixture);
+    ws = spectator.inject(WebSocketService);
+    spectator.fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('should show table headers', async () => {
+    const table = await loader.getHarness(IxTableHarness);
+    const headerRow = await table.getHeaderRow();
+
+    expect(headerRow).toMatchObject({
+      name: 'Name',
+      state: 'Running',
+      enable: 'Start Automatically',
+      actions: '',
+    });
+  });
+
+  it('should show table rows', async () => {
+    const table = await loader.getHarness(IxTableHarness);
+    const cells = await table.getCells(true);
+    const expectedData = [...serviceNames.keys()]
+      .filter((service) => ![ServiceName.Gluster, ServiceName.Afp].includes(service))
+      .map((service) => [serviceNames.get(service), '', '', 'edit']);
+    const expectedRows = [['Name', 'Running', 'Start Automatically', ''], ...expectedData];
+
+    expect(cells).toEqual(expectedRows);
+  });
+
+  it('should redirect to configure service page', async () => {
+    const editButton = await loader.getHarness(MatButtonHarness.with({ text: 'edit' }));
+    await editButton.click();
+
+    expect(spectator.inject(Router).navigate).toHaveBeenCalledWith(['/services', 'dynamicdns']);
+  });
+
+  it('should check service state changes', async () => {
+    const slideToggle = await loader.getHarness(MatSlideToggleHarness);
+    await slideToggle.toggle();
+
+    expect(await slideToggle.isChecked()).toBeTruthy();
+    expect(ws.call).toHaveBeenCalledWith('service.start', ['dynamicdns']);
+  });
+
+  it('should check service autostart changes', async () => {
```

Comment:
I'd prefer something like 'should change service autostart state when checkbox is ticked'.
Same for one above.